server/src/main/scala/org/http4s/server/staticcontent/WebjarService.scala

```diff
@@ -88,21 +88,33 @@ object WebjarService {
   /**
     * Returns an Option(WebjarAsset) for a Request, or None if it couldn't be mapped
     *
-    * @param subPath The request path without the prefix
+    * @param p The request path without the prefix
     * @return The WebjarAsset, or None if it couldn't be mapped
     */
   private def toWebjarAsset(p: Path): Option[WebjarAsset] = {
     val count = p.getNameCount
     if (count > 2) {
       val library = p.getName(0).toString
       val version = p.getName(1).toString
-      val asset = p.subpath(2, count)
-      Some(WebjarAsset(library, version, asset.toString))
+      val asset = asScalaIterator(p.subpath(2, count).iterator()).mkString("/")
+      Some(WebjarAsset(library, version, asset))
     } else {
       None
     }
   }
 
+  /** Creates a scala.Iterator from a java.util.Iterator.
+    *
+    * We're not using scala.jdk.CollectionConverters (which was added in 2.13)
+    * or scala.collection.convert.ImplicitConversion (which was deprecated in 2.13)
+    * to ease cross-building against multiple Scala versions.
+    */
+  private def asScalaIterator[A](underlying: java.util.Iterator[A]): Iterator[A] =
+    new Iterator[A] {
+      override def hasNext: Boolean = underlying.hasNext
+      override def next(): A = underlying.next()
+    }
+
```

Comment:
This could probably be moved to `CollectionCompat`, in case it's useful elsewhere, but I don't think that's introduced until 0.21.

Reply:
Oh, I didn't noticed `CollectionCompat`. It exists on the `series/0.20` branch. On `series/0.21` it also has a `CollectionConverters` that could be used here.